romancal/skycell/skymap.py

```diff
@@ -0,0 +1,635 @@
+            (
+                self.data[["ra_corn1", "dec_corn1"]].item(),
+                self.data[["ra_corn2", "dec_corn2"]].item(),
+                self.data[["ra_corn3", "dec_corn3"]].item(),
+                self.data[["ra_corn4", "dec_corn4"]].item(),
+            )
+        )
+
+    @cached_property
+    def vectorpoint_corners(self) -> NDArray[float]:
+        """corners in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(
+            np.stack(sgv.lonlat_to_vector(*np.array(self.radec_corners).T), axis=1)
+        )
+
+    @cached_property
+    def vectorpoint_center(self) -> tuple[float, float, float]:
+        """center in 3D Cartesian space on the unit sphere"""
+        return sgv.normalize_vector(np.array(sgv.lonlat_to_vector(*self.radec_center)))
+
+    @cached_property
+    def polygon(self) -> sgp.SingleSphericalPolygon:
+        """spherical polygon representing this sky cell"""
+        return sgp.SingleSphericalPolygon(
+            points=self.vectorpoint_corners,
+            inside=self.vectorpoint_center,
+        )
+
+    @cached_property
+    def projection_region(self) -> "ProjectionRegion":
+        """Projection region containing this sky cell."""
+        if self.index is None:
+            raise ValueError("no index provided")
+        return ProjectionRegion.from_skycell_index(self.index)
+
+    @property
+    def pixel_scale(self) -> float:
+        """degrees per pixel"""
+        return self._skymap.pixel_scale
+
+    @property
+    def pixel_shape(self) -> tuple[int, int]:
+        """number of pixels across"""
+        return self._skymap.pixel_shape
+
+    @property
+    def wcs_info(self) -> dict[str, float | str]:
+        """WCS properties as defined in the Level 3 association schema"""
+
+        return {
+            "name": self.name,
+            "pixel_scale": self.pixel_scale,
+            "ra_projection_center": self.projection_region.radec_tangent[0],
+            "dec_projection_center": self.projection_region.radec_tangent[1],
+            "x0_projection": self.xy_tangent[0],
+            "y0_projection": self.xy_tangent[1],
+            "ra_center": self.radec_center[0],
+            "dec_center": self.radec_center[1],
+            "nx": self.pixel_shape[0],
+            "ny": self.pixel_shape[1],
+            "orientat": self.orientation,
+            "orientat_projection_center": self.projection_region.orientation
+            if self.projection_region.data["dec_max"] != 90.0
+            # rotate north polar cap by 180 degrees
+            # TODO: find out why this is necessary...
+            else self.projection_region.orientation + 180,
+        }
+
+    @cached_property
+    def wcs(self) -> WCS:
+        """WCS representing this sky cell"""
+        wcsobj = wcsinfo_to_wcs(
+            self.wcs_info,
+            bounding_box=(
+                (-0.5, self.pixel_shape[0] + 0.5),
+                (-0.5, self.pixel_shape[1] + 0.5),
```

Comment:
I always screw up the conventions here.  Is this right?  I expect this to roughly be the very top of the image to the very bottom of the image.  I think that should contain pixel.shape pixels; i.e., I expect bounding_box[0][1] - bounding_box[0][0] = pixel.shape[0], but here it's pixel.shape[0] + 1.  So a 1x1 pixel image will have a bounding box covering a 2x2 pixel region??